symengine/rational.cpp

```diff
@@ -91,4 +91,68 @@ void get_num_den(const Rational &rat,
     *num = integer((rat.i).get_num());
     *den = integer((rat.i).get_den());
 }
+
+bool Rational::is_perfect_power(bool is_expected) const
+{
+    const mpz_class &num = i.get_num();
+    if (mpz_cmp_ui(num.get_mpz_t(), 0) == 0)
+        return true;
+    else if (mpz_cmp_ui(num.get_mpz_t(), 1) == 0)
+        return mpz_perfect_power_p(i.get_den().get_mpz_t()) != 0;
+
+    const mpz_class &den = i.get_den();
+    if (mpz_cmp_ui(den.get_mpz_t(), 1) == 0)
+        return mpz_perfect_power_p(num.get_mpz_t()) != 0;
+    
+    if (not is_expected) {
+        if (mpz_cmpabs(num.get_mpz_t(), den.get_mpz_t()) > 0) {
+            if (mpz_perfect_power_p(den.get_mpz_t()) == 0)
+                return false;
+        }
+        else {
+            if (mpz_perfect_power_p(num.get_mpz_t()) == 0)
+                return false;
+        }
+    }
+    mpz_t prod;
```

Comment:
I would suggest to use `mpz_class`. Then you don't have to worry about memory leaks, as you find out, it is so easy to leak memory.